server/src/main/java/org/apache/druid/server/coordinator/helper/NewestSegmentFirstIterator.java

```diff
@@ -229,15 +236,58 @@ public boolean hasNext()
     }
   }
 
+  private static boolean needsCompaction(DataSourceCompactionConfig config, SegmentsToCompact candidates)
```

Comment:
Which tests cover this new logic?

Reply:
`DruidCoordinatorSegmentCompactorTest` verifies the behavior of `DruidCoordinatorSegmentCompactor`.